v-next/hardhat/src/internal/builtin-plugins/solidity/config.ts

```diff
@@ -300,12 +309,17 @@ function resolveSolidityConfig(
           },
         ),
       ),
+      preferWasm,
     };
   }
 
+  // This will generate default build profiles (e.g. production) when they are not specified in the config, cloning from 'default', which is always present
   for (const profile of DEFAULT_BUILD_PROFILES) {
     if (!(profile in profiles)) {
-      profiles[profile] = profiles.default;
+      profiles[profile] = {
+        ...profiles.default,
```

Comment:
Unrelated to this particular PR, but we should probably do a deep copy here. Otherwise this happens:

<img width="620" height="180" alt="Image" src="https://github.com/user-attachments/assets/0e511be0-2da3-43e7-900d-f54fec26f9b2" />

It seems pedantic, but we've been bitten by this before.

But let's not do it in this PR. We should have a separate issue to make sure that both the user and the resolved config are made up of independent objects.